File: app/src/main/java/app/deadmc/devnetworktool/models/RestRequestHistory.java
package app.deadmc.devnetworktool.models;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.orm.SugarRecord;

import java.lang.reflect.Type;
import java.util.ArrayList;

/**
 * Created by adanilov on 22.03.2017.
 */

public class RestRequestHistory extends SugarRecord {
    private long timeLastUsage;
    private String url;
    private String method;
    private String headers;
    private String requests;

    public RestRequestHistory() {

    }

    public RestRequestHistory(String url, String method, ArrayList<KeyValueModel> headers,ArrayList<KeyValueModel> requests) {
        Gson gson = new Gson();
        this.timeLastUsage = System.currentTimeMillis();
        this.url = url;
        this.method = method;
        this.headers = gson.toJson(headers);
        this.requests = gson.toJson(requests);
    }

    public long getTimeLastUsage() {
        return timeLastUsage;
    }

    public String getUrl() {
        return url;
    }

    public String getMethod() {
        return method;
    }

    public ArrayList<KeyValueModel> getHeaders() {
        return getArrayList(headers);
    }

    public ArrayList<KeyValueModel> getRequests() {
        return getArrayList(requests);
    }

    private ArrayList<KeyValueModel> getArrayList(String stringValue) {
        if (stringValue.isEmpty()) return new ArrayList<>();
        Type type = new TypeToken<ArrayList<KeyValueModel>>(){}.getType();
        Gson gson = new Gson();
        ArrayList<KeyValueModel> arrayList = gson.fromJson(stringValue, type);
        return arrayList;
    }




}

File: app/src/main/java/app/deadmc/devnetworktool/models/RestRequestHistory.kt
package app.deadmc.devnetworktool.models

import app.deadmc.devnetworktool.interfaces.model.BaseModel
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import com.orm.SugarRecord

import java.lang.reflect.Type
import java.util.ArrayList

class RestRequestHistory : SugarRecord,BaseModel {
    var timeLastUsage: Long = System.currentTimeMillis()
    var url: String = ""
    var method: String = ""
    private var headers: String = ""
    private var requests: String = ""

    constructor()

    constructor(url: String, method: String, headers: ArrayList<KeyValueModel>, requests: ArrayList<KeyValueModel>) {
        this.url = url
        this.method = method
        this.headers = Gson().toJson(headers)
        this.requests = Gson().toJson(requests)
    }

    fun getHeaders(): ArrayList<KeyValueModel> {
        return getArrayList(headers)
    }

    fun getRequests(): ArrayList<KeyValueModel> {
        return getArrayList(requests)
    }

    override fun getUniqueId(): String {
        return id.toString()
    }

    private fun getArrayList(stringValue: String): ArrayList<KeyValueModel> {
        if (stringValue.isEmpty()) return ArrayList()
        val type = object : TypeToken<ArrayList<KeyValueModel>>() {

        }.type
        val gson = Gson()
        return gson.fromJson(stringValue, type)
    }


}
